File: Top 100 Codes/06-Advanced Codes related to Arrays/22-Find longest consecutive subsequence/Findlongestconsecutivesubsequence.py
def find_longest_consecutive_subsequence(nums):
    num_set = set(nums)
    longest_streak = 0

    for num in num_set:
        # Only check if it's the start of a sequence
        if num - 1 not in num_set:
            current_num = num
            current_streak = 1

            # Count up the consecutive numbers
            while current_num + 1 in num_set:
                current_num += 1
                current_streak += 1

            longest_streak = max(longest_streak, current_streak)
    
    return longest_streak

# Example usage
nums = [100, 4, 200, 1, 3, 2]
print("Longest Consecutive Subsequence Length:", find_longest_consecutive_subsequence(nums))

File: Top 100 Codes/06-Advanced Codes related to Arrays/22-Find longest consecutive subsequence/Findlongestconsecutivesubsequence.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
using namespace std;

int findLongestConsecutiveSubsequence(vector<int>& nums) {
    unordered_set<int> numSet(nums.begin(), nums.end());
    int longestStreak = 0;

    for (int num : numSet) {
        // Check if it's the start of a sequence
        if (numSet.find(num - 1) == numSet.end()) {
            int currentNum = num;
            int currentStreak = 1;

            // Count up
            while (numSet.find(currentNum + 1) != numSet.end()) {
                currentNum++;
                currentStreak++;
            }

            longestStreak = max(longestStreak, currentStreak);
        }
    }
    return longestStreak;
}

int main() {
    vector<int> nums = {100, 4, 200, 1, 3, 2};
    cout << "Longest Consecutive Subsequence Length: " << findLongestConsecutiveSubsequence(nums) << endl;
    return 0;
}
